### PR TITLE
Add sbt-version-policy

### PR DESCRIPTION
**Pull request by julienrf**
Initially, my goal was to declare the versioning scheme used by zio. sbt-version-policy does that, and it also checks that your version numbers match the declared versioning scheme.

The benefit of declaring the versioning scheme used by zio is that all the users of zio will benefit from accurate diagnostics of conflicting dependencies.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -57,9 +57,9 @@ jobs:
         java-version: ${{ matrix.java }}
     - name: Cache scala dependencies
       uses: coursier/cache-action@v6
-    - name: Mima Checks
-      if: ${{ !startsWith(matrix.scala, '2.11.') && !startsWith(matrix.scala, '3.0.') && matrix.platform == 'JVM' }}
```
Comment (julienrf) [inline .github/workflows/ci.yml:61]: compatibility checks will also run on Scala 3

**Review by adamgfraser — DISMISSED** [review]
@julienrf This looks great! Thank you!

File: project/plugins.sbt [hunk 2]
```diff
@@ -1,10 +1,10 @@
 addSbtPlugin("ch.epfl.scala"                     % "sbt-bloop"                     % "1.4.8")
 addSbtPlugin("ch.epfl.scala"                     % "sbt-scalafix"                  % "0.9.29")
+addSbtPlugin("ch.epfl.scala"                     % "sbt-version-policy"            % "1.2.1")
 addSbtPlugin("com.eed3si9n"                      % "sbt-buildinfo"                 % "0.10.0")
 addSbtPlugin("com.geirsson"                      % "sbt-ci-release"                % "1.5.7")
 addSbtPlugin("com.github.cb372"                  % "sbt-explicit-dependencies"     % "0.2.16")
 addSbtPlugin("com.thoughtworks.sbt-api-mappings" % "sbt-api-mappings"              % "3.0.0")
-addSbtPlugin("com.typesafe"                      % "sbt-mima-plugin"               % "0.8.1")
```
Comment (julienrf) [inline project/plugins.sbt:7]: sbt-version-policy already includes sbt-mima-plugin.

However, it currently depends on version 0.9.2, which is known to suffer from https://github.com/lightbend/mima/issues/639 (which makes this PR fail). We will probably have to wait for 0.9.3…

**Comment by jdegoes** [comment]
@julienrf Would love to get this in! Any idea when you'll have a chance to fix the build issues??

**Comment by julienrf** [comment]
May I ask a maintainer to approve running the workflow?

**Comment by adamgfraser** [comment]
@julienrf Done!

**Comment by julienrf** [comment]
Thank you Adam. I see a failure but I don’t think it’s related to my PR. Could you please check?

**Comment by jdegoes** [comment]
@julienrf This looks to have bit-rotted. Would still love to get this in if you have any spare cycles to work on it!

Will close for now, please re-open whenever you'd like a 2nd look.

**Pull request closed without merging**